.github/workflows/mypy_primer.yml

```diff
@@ -0,0 +1,99 @@
+name: Run mypy_primer
+
+on:
+  # Only run on PR, since we diff against main
+  pull_request:
+    paths:
+      - "**/*.pyi"
+      - ".github/workflows/mypy_primer.yml"
+      - ".github/workflows/mypy_primer_comment.yml"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  mypy_primer:
+    name: Run
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        shard-index: [0]  # e.g. change this to [0, 1, 2] and --num-shards below to 3
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
We like to have commit hashes on these. You can copy from other github jobs for the common actions, but some here are new to us.